File: packages/api/src/entity/CoordinateOptional.ts
import { Column, EntitySchema } from "typeorm";
import { type ICoordinateOptional } from "models";

// Embeddable Coordinate columns that are nullable
const CoordinateOptional = new EntitySchema<ICoordinateOptional>({
  name: "coordinate",
  columns: {
    lat: {
      type: "decimal",
      nullable: true,
    },
    lng: {
      type: "decimal",
      nullable: true,
    },
  },
});

export default CoordinateOptional;

File: go-server/internal/models/coordinate_optional.go
package models

type CoordinateOptional struct {
	Lat *float64 `json:"lat,omitzero" sqlp:"lat"`
	Lng *float64 `json:"lng,omitzero" sqlp:"lng"`
}

func (c CoordinateOptional) IsZero() bool {
	return c.Lat == nil && c.Lng == nil
}
